mllib-dal/src/main/scala/com/intel/oap/mllib/OneDAL.scala

```diff
@@ -74,7 +74,7 @@ object OneDAL {
   }
 
   def isDenseDataset(ds: Dataset[_]): Boolean = {
-    val row = ds.select("features").head()
+    val row = ds.select(new Column("features")).head()
```

Comment:
@minmingzhu Your change is the same as the previous one. You need to get the real feature column using featuresCol param instead. For instance: https://spark.apache.org/docs/latest/api/scala/org/apache/spark/ml/classification/NaiveBayes.html#featuresCol:org.apache.spark.ml.param.Param[String]
And pass the featureColName as parameter from the algorithm class.